source/crud/query/cursors.txt

```diff
@@ -0,0 +1,178 @@
+.. _csharp-cursors:
+
+=========================
+Access Data From a Cursor
+=========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: read, results, oplog
+
+Overview
+--------
+
+In this guide, you can learn how to access data from a **cursor** by using the
+{+driver-short+}.
+
+A cursor is a mechanism that returns the results of a read operation in iterable
+batches. Because a cursor holds only a subset of documents at any given time,
+cursors reduce both memory consumption and network bandwidth usage.
+
+You can retrieve a cursor by using the ``FindSync()`` and ``FindAsync()`` methods. You can
+also convert the results of the ``Find()`` method to a cursor by chaining the ``ToCursor()``
+or ``ToCursorAsync()`` method.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection
+in the ``sample_restaurants`` database provided in the :atlas:`Atlas sample datasets </sample-data>`.
+To learn how to create a free MongoDB Atlas cluster and load the sample datasets,
+see the :ref:`<csharp-get-started>` tutorial.
+
+The examples on this page use the following ``Restaurant`` object to model the documents
+in the ``restaurants`` collection:
+
+.. literalinclude:: /includes/fundamentals/code-examples/Cursor.cs
+   :start-after: start-restaurant-class
+   :end-before: end-restaurant-class
+   :language: csharp
+
+.. _csharp-cursors-iterate:
+
+Access Cursor Contents Iteratively
+----------------------------------
+
+To iterate over the contents of a cursor, use a ``foreach`` loop inside a ``using`` block.
+The following example retrieves all documents from the ``restaurants`` collection and
```

Comment:
S: consider showing an example with a filter so that users dont try this code and potentially get an error from the amount of documents